### ISSUE TITLE
PDF Watermark AND XRechnung (ZUGFeRD XML)

### ISSUE THREAD

**Issue by gklimm** [InvoicePlane/InvoicePlane#912]
## Expected Behavior

Generate the PDF

## Current Behavior

Exception (see below)

## Steps to Reproduce

1. Switch on "ZUGFeRD XML"
2. Activate Watermarks in PDF generation
3. Try to generate a PDF of an invoice

## Failure Logs

An uncaught Exception was encountered
Type: Mpdf\MpdfException

Message: PDFA and PDFX do not permit transparency, so mPDF does not allow Watermarks!

Filename: /var/www/ip/vendor/mpdf/mpdf/src/Mpdf.php

Line Number: 10546

Backtrace:

File: /var/www/ip/vendor/mpdf/mpdf/src/Mpdf.php
Line: 13205
Function: watermark

File: /var/www/ip/vendor/mpdf/mpdf/src/Mpdf.php
Line: 2054
Function: Footer

File: /var/www/ip/vendor/mpdf/mpdf/src/Mpdf.php
Line: 9519
Function: Close

File: /var/www/ip/application/helpers/mpdf_helper.php
Line: 115
Function: Output

File: /var/www/ip/application/helpers/pdf_helper.php
Line: 105
Function: pdf_create

File: /var/www/ip/application/modules/invoices/controllers/Invoices.php
Line: 272
Function: generate_invoice_pdf

File: /var/www/ip/index.php
Line: 329
Function: require_once

**Comment by Kenny1291** [comment]
It's a mpdf limitation.
https://github.com/mpdf/mpdf/issues/1625

**Comment by gklimm** [comment]
In itself, there is nothing wrong with not being able to do this. It should just not be selectable for the user in the settings. The two settings are then already XOR in the interface.

**Comment by Verony-makesIT** [comment]
It is, as "Kenny1291" points out, a problem in mpdf.
Programming and or adjusting something (_your settings selection xor proposal_) in IP only to then, later or much later (?), maybe have to adjust it again when the problem in mpdf is solved, I personally don't find very productive. 
I think it is better to use what is available in IP and mpdf today to achieve your goal (**embedding Zugferd XML + paid/overdue stamp on your invoice**).
This can be done by using the following workaround:
- [ ] Deactivate the "watermark" setting and activate the "Zugferd" setting. 
- [ ] In your (paid or overdue) invoice template you can then, in the places that you want, add your "paid/overdue" stamp.
- [ ] In my "invoice - paid" template, I added (instead of the watermark) and before the footer tag the following code:

**div img:**
```
<div style="text-align: right; padding: 0; margin: 0;">
    <?php echo '<img src="' . base_url('assets/core/img/paid.png') . '" class="paid-stamp">'; ?>
</div> 
<!--<watermarktext content="<?php _trans('paid'); ?>" alpha="0.3" />-->
```
**style class paid-stamp:**   
```
<style>
.paid-stamp {
    position: absolute;
    margin-top: -32.5px;
    margin-right: 300px; 
    width: 100px;
    transform: rotate(22deg);
}
</style>  
```

**Here you can see the result of my suggestion:**
![image](https://user-images.githubusercontent.com/30659226/221009782-fe1750b7-ce65-4288-9d09-ebcc5ee9bb91.png)

Remark:
However, I am not a specialist in the _Zugferd version in IP_ but in my opinion it is somewhat outdated and not really compatible with the **latest Zugferd 2.1.1 version and with the European EN 16931 standard**...

_I hope I have been of assistance to you with this..._

**Comment by nielsdrost7** [comment]
@gklimm
> it should just not be selectable for the user in the settings

I agree. What would you suggest? Apply Verony's changes in your local template?

**Comment by nielsdrost7** [comment]
Solved by creating and releasing version 1.6.3

@sudwebdesign this one is resolved, correct?
